src/cache/mod.rs

```diff
@@ -0,0 +1,12 @@
+mod redis;
+pub use redis::RedisCache;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use std::marker::{Send, Sync};
+
+#[async_trait]
+pub trait Cache<T>: Send + Sync {
+    async fn set<S: ToString + Send + Sync>(id: S, obj: T) -> Result<()>;
```

Comment:
add an extra argument `timeout` or `ttl`

Reply:
I remember that we said it is in the twin it self, so that we don't need to add it as a separate parameter